[Docs] Add Encoder requirements to PR Checklist doc (#17634)
qmk/qmk_firmware: docs/pr_checklist.md

    - `matrix_init_board()` etc. migrated to `keyboard_pre_init_kb()`, see: [keyboard_pre_init*](custom_quantum_functions.md?id=keyboard_pre_init_-function-documentation)
    - prefer `CUSTOM_MATRIX = lite` if custom matrix used, allows for standard debounce, see [custom matrix 'lite'](custom_matrix.md?id=lite)
    - prefer LED indicator [Configuration Options](feature_led_indicators.md?id=configuration-options) to custom `led_update_*()` implementations where possible

>     - Encoder support should not be hacked into the keymap here.  (eg, no `tap_code(dynamic_keymap_get_keycode())` or `action_exec()` hacks).  The [Encoder Map](feature_encoders.md?id=encoder-map) feature already supports the dynamic keymap feature (what power's VIA/VIAL's "live keymaps").

Reviewer: ```suggestion
    - Encoder support should not be hacked into the keymap here -- no `tap_code(dynamic_keymap_get_keycode())` or `action_exec()` hacks.  The [Encoder Map](feature_encoders.md?id=encoder-map) feature already supports the dynamic keymap feature (what power's VIA's "live keymap updates" capability). VIA has an open issue to support this already [here](https://github.com/the-via/app/issues/26).
```

>       - If support is absolutely necessary, it should be implemented exclusively at the keymap level, with none of the implementation bleeding into the keyboard level (no empty rows/columns, no encoder specific layouts, etc), as such configuration can be redefined at the keymap level.  And should use the `action_exec` hack. <!-- because people will complain, give them a way to implement it, in the meanwhile.  To be removed. -->

Reviewer: ```suggestion
      - If support is absolutely necessary, it should be implemented exclusively at the keymap level, with none of the implementation bleeding into the keyboard level (no empty rows/columns, no encoder specific layouts, etc.), as those configurations can be redefined at the keymap level. Keymaps can then choose to use the `action_exec` hack. <!-- because people will complain, give them a way to implement it, in the meanwhile.  To be removed. -->
```
